### DISCUSSION TITLE
Unable to save REST API Authetications

### DISCUSSION THREAD

**Discussion by natera** [Budibase/budibase#10979] [Help] [answered]
I'm running a self-hosted budibase  ( 2.7.21 )  in docker.

When I add an Authentication to a REST API source and publish the changes, the authentication does not get saved.

I have tried the solution of this #8498, but I have the same result, it does not get saved

**Comment by natera** [comment]
![Peek 2023-06-21 22-41](https://github.com/Budibase/budibase/assets/4206506/7aaa69cb-15c9-4977-b274-cff41cfd06ba)

Here is a gif of the error

**Reply by ConorWebb96** [reply, marked as answer]
Hey @natera,

I've had a look at this, I noticed in your gif you didn't save the settings at the top of this configuration before clicking the publish button and refreshing. I've attached a screenshot of the button you need to click when you make changes below;
<img width="646" alt="Screenshot 2023-06-22 at 08 46 55" src="https://github.com/Budibase/budibase/assets/126772285/e1d66d7a-722b-4bb1-817a-52eff206590a">

Any changes within the area you've talked about need to be saved using that button, otherwise anything entered will be lost on a page refresh. I wasn't able to replicate after hitting the save button. All settings where saved and appear after publishing and refreshing for myself.

I hope this helps!

**Reply by natera** [reply]
Thanks @ConorWebb96 this solves my issue
